src/azure-cli-core/azure/cli/core/telemetry.py

```diff
@@ -414,6 +416,18 @@ def set_raw_command_name(command):
     _session.raw_command = command
 
 
+@decorators.suppress_all_exceptions()
+def add_dedicated_instrumentation_key(dedicated_instrumentation_key):
+    if not dedicated_instrumentation_key:
+        return
+
+    from collections.abc import Iterable
+    if isinstance(dedicated_instrumentation_key, str):
+        _session.instrumentation_key.add(dedicated_instrumentation_key)
+    elif isinstance(dedicated_instrumentation_key, Iterable):
+        _session.instrumentation_key.update(dedicated_instrumentation_key)
```

Comment:
Just out of curiosity, may I ask can we support adding multiple `dedicated_instrumentation_key`?

Reply:
Just in case they want to send to several appInsights

Reply:
Got it, thanks~